src/components/application_manager/src/rpc_service_impl.cc

```diff
@@ -535,13 +527,58 @@ void RPCServiceImpl::SendMessageToMobile(
                (*message)[jhs::S_PARAMS][jhs::S_MESSAGE_TYPE].asInt())) {
     allow_unknown_parameters = false;
   }
+
+  const auto api_function_id = static_cast<mobile_apis::FunctionID::eType>(
+      (*message)[strings::params][strings::function_id].asUInt());
+
+  if (app) {
```

Comment:
@YarikMamykin looks like app pointer was already checked previously.

Reply:
@LuxoftAKutsan yes, app pointer was already checked but no actions like return from method are permitted, so there is still potential danger of having a non-valid pointer.